### PR TITLE
hirte-agent crashes if config files are not present

### PR DESCRIPTION
**Pull request by rhatdan**
If you call cgd_dispose then you need to null the value.

### CODE REVIEW

File: src/agent/agent.c [hunk 1]
```diff
@@ -481,6 +481,7 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
                         CFG_ETC_AGENT_CONF_DIR);
         if (result != 0) {
                 cfg_dispose(agent->config);
+                agent->config=NULL;
```
Comment (mwperina) [inline src/agent/agent.c]: Hmm, why is this needed? After deallocating the memory we are exiting the program immediately at
https://github.com/containers/hirte/blob/main/src/agent/main.c#L128 so we should not be accessing the property at all. Or am I missing something?

Comment (rhatdan) [reply src/agent/agent.c]: The cleanup of the system calls
 cfg_dispose(agent->config) a second time causing a crash.

If you remove the config from the system you will see the error followed by a segfault.

**Comment by rhatdan** [comment]
What is the linter complaining about? When I run the linter locally  I get the linter blowing up.
```
markdownlint-cli2 | echo "markdownlint-cli2 not found, skipping markdown lint"
markdownlint-cli2 not found, skipping markdown lint
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:930:11)
    at Socket._write (node:net:942:8)
    at writeOrBuffer (node:internal/streams/writable:392:12)
    at _write (node:internal/streams/writable:333:10)
    at Writable.write (node:internal/streams/writable:337:10)
    at console.value (node:internal/console/constructor:300:16)
    at console.log (node:internal/console/constructor:377:26)
    at main (/usr/local/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:859:5)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}

Node.js v18.15.0
```

But no report of an error.  I also have markdownlint-cli2 installed, but it still reports it is missing.

**Comment by mwperina** [comment]
In the build job logs you see following lines

```
ninja -C builddir clang-format-check
ninja: Entering directory `builddir'
[0/1] /usr/bin/meson --internal clangformat /__w/hirte/hirte /__w/hirte/hirte/builddir --check
File reformatted:  /__w/hirte/hirte/src/agent/agent.c
FAILED: meson-internal__clang-format-check 
```

So there is a formatting problem in `agent.c`.
When you run `make fmt` locally, following changes are performed by formatter:

```diff
diff --git a/src/agent/agent.c b/src/agent/agent.c
index 693e347..83239ef 100644
--- a/src/agent/agent.c
+++ b/src/agent/agent.c
@@ -481,7 +481,7 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
                         CFG_ETC_AGENT_CONF_DIR);
         if (result != 0) {
                 cfg_dispose(agent->config);
-                agent->config=NULL;
+                agent->config = NULL;
                 return false;
         }
```

File: src/agent/agent.c (hunk 1 above)
Comment (mwperina) [reply src/agent/agent.c]: Ah, right, thanks for explanation

**Review by mwperina — APPROVED** [review]
+1
